homeassistant/helpers/trigger_template_entity.py

```diff
@@ -74,6 +103,50 @@ def make_template_entity_base_schema(default_name: str) -> vol.Schema:
 ).extend(TEMPLATE_ENTITY_BASE_SCHEMA.schema)
 
 
+class ValueTemplate(Template):
+    """Class to hold a value_template and manage caching and rendering it with 'value' in variables."""
+
+    @classmethod
+    def from_template(cls, template: Template) -> ValueTemplate:
+        """Create a ValueTemplate object from a Template object."""
+        return cls(template.template, template.hass)
+
+    @callback
+    def async_render_as_value_template(
+        self,
+        entity_id: str,
+        variables: dict[str, Any],
+        error_value: Any = _SENTINEL,
+    ) -> Any:
+        """Render template that requires 'value' and optionally 'value_json'.
+
+        Template errors will be supprested when an error_value is supplied.
+
+        This method must be run in the event loop.
+        """
+        self._renders += 1
+
+        if self.is_static:
+            return self.template
+
+        compiled = self._compiled or self._ensure_compiled()
+
+        try:
+            render_result = _render_with_context(
+                self.template, compiled, **variables
+            ).strip()
+        except jinja2.TemplateError as ex:
+            message = f"Error parsing value for {entity_id}: {ex} (value: {variables['value']}, template: {self.template})"
+            logger = logging.getLogger(f"{__package__}.{entity_id.split('.')[0]}")
+            if error_value is _SENTINEL:
+                logger.error(message)
+            else:
+                logger.debug(message)
+            return variables["value"] if error_value is _SENTINEL else error_value
```

Comment:
The non-forgiving case is only used in tests AFAICT, I suggest to simplify the implementation like this.
Maybe mention in the docstring that support for a non-forgiving error value could be added if needed?
```suggestion
            logger.debug(message)
            return error_value
```